ui/widgets/nuxeo-html-editor.js

```diff
@@ -55,121 +51,101 @@ import './alloy/alloy-ocean.js';
-                    'strike',
-                    'paragraphLeft',
-                    'paragraphCenter',
-                    'paragraphJustify',
-                    'ul',
-                    'ol',
-                    'quote',
-                    'link',
-                    'removeFormat',
-                  ],
-                  test: AlloyEditor.SelectionTest.text,
-                },
-                {
-                  name: 'table',
-                  buttons: ['tableRow', 'tableColumn', 'tableCell', 'tableRemove'],
-                  getArrowBoxClasses: AlloyEditor.SelectionGetArrowBoxClasses.table,
-                  setPosition: AlloyEditor.SelectionSetPosition.table,
-                  test: AlloyEditor.SelectionTest.table,
-                },
-              ],
-              tabIndex: 1,
-            },
-          };
-      const editor = AlloyEditor.editable(this.$.editor, {
-        readOnly: this.readOnly,
-        uiNode: this.root,
-        toolbars,
+      const toolbar = this.hideToolbars
+        ? []
+        : [
+            [{ header: [1, 2, 3, 4, 5, 6, false] }], // headings
+            ['bold', 'italic', 'underline', 'strike'], // style buttons
+            [{ align: '' }, { align: 'center' }, { align: 'right' }], // text alignment
+            [{ color: [] }, { background: [] }], // dropdown with defaults from theme
+            [{ script: 'sub' }, { script: 'super' }], // superscript/subscript
+            ['blockquote'], // blockquote
+            [{ list: 'ordered' }, { list: 'bullet' }], // lists
+            ['link', 'image'], // links and media
+            ['clean'], // remove formatting button
+          ];
+
+      const { placeholder, readOnly } = this;
+      this._editor = new Quill(this.$.editor, { theme: 'snow', modules: { toolbar }, placeholder, readOnly });
+
+      // update value on change
+      this._editor.on('text-change', () => {
+        this._debouncer = Debouncer.debounce(this._debouncer, timeOut.after(200), () => this._updateValue());
       });
-      this.element = editor.get('nativeEditor');
-      // set initial value
-      this.element.setData(this.value);
-
-      this.element.on('change', this._updateValue.bind(this));
-
-      this.element.on('actionPerformed', this._updateValue.bind(this));
     }
 
     _updateValue() {
       this._internalChange = true;
-      this.value = this.element.getData();
+      this.value = this._editor.root.innerHTML;
       this._internalChange = false;
     }
 
     _valueChanged() {
-      if (this.element && !this._internalChange) {
-        this.element.setData(this.value);
+      if (this._editor && !this._internalChange) {
+        const delta = this._editor.clipboard.convert({ html: this.value });
+        this._editor.setContents(delta, Quill.sources.USER);
+        this._editor.setSelection(0, Quill.sources.SILENT);
       }
     }
+
+    get _toolbar() {
+      return this._editor.getModule('toolbar');
+    }
```

Comment:
Not sure if this brings value since it is only being used once, correct?

Reply:
was mostly for convenience as i used it also in the tests. dropped